### PR TITLE
Payment Methods: Add toggle for backup payment methods

### PR DESCRIPTION
**Pull request by sirbrillig**
#### Changes proposed in this Pull Request

This is part of a project to allow marking payment methods as available for "backup" which means that the payment method may be used if another payment method fails during an auto-renew. This PR adds a toggle to each payment method in the list of payment methods to allow toggling that method's "backup" status.

Depends on D67557-code which adds the appropriate endpoints.

#### Screenshots

Screen Shot 2021-09-30 at 6 56 21 PM

Screen Shot 2021-09-30 at 7 11 42 PM

#### Testing instructions

- Visit `/me/purchases/payment-methods` (or the similar site-level page; they both use the same components).
- Verify that you see some indication that each payment method is not marked as a backup.
- Click to change a payment method to backup.
- Verify that the payment method changes to indicate that it's now a backup.
- Reload the page and verify that the payment method remains a backup.

### CODE REVIEW

**Comment by pottedmeat** [comment]
Use https://wordpress.com/support/payment/#manage-payment-methods for the Learn More link.

**Comment by martian77** [comment]
Using firefox, I checked the box and left it with the focus box around it (see picture). Refreshing the page mostly(!) did not correctly load in the data. Mostly (again, !) when I click away the page refresh reloaded as I expected.

focus-highlighted-checkbox

I'm not really convinced this is related to the focus box - it feels like it might be a time thing, but I've tried leaving it for longer periods and sometimes it still isn't saved. Not quite sure if it's something my end or not!

**Comment by sirbrillig** [comment]
> Using firefox, I checked the box and left it with the focus box around it (see picture). Refreshing the page mostly(!) did not correctly load in the data. Mostly (again, !) when I click away the page refresh reloaded as I expected.

🤔 I wonder if this has to do with the optimistic update... perhaps the update endpoint is not working that well or is very slow?

Alternatively, it could be a caching issue; the data used by the page's checkboxes initially comes from the Redux store's payment method list. The code in this PR updates that list, but maybe it's not working right. I had to jump through hoops to keep react-query from fetching the data itself for every checkbox, but if the cached data is wrong, we could always allow it to make those queries or force an update of the full payment method list.

Now that the endpoints are merged, could you try again with your devtools showing network requests so we can see if the changes are not being persisted or if this is purely a cache thing.

**Comment by martian77** [comment]
> Now that the endpoints are merged, could you try again with your devtools showing network requests so we can see if the changes are not being persisted or if this is purely a cache thing.

I tried this again today, and I can't recreate it. Caching, or maybe a slow sandbox? Either way, it seems to be working fine now.

**Comment by sirbrillig** [comment]
Thanks! I'll mark this as "reviewed" for right now, but I think we shouldn't merge it until we have finished adding support for using these backup payment methods when subscriptions fail on the backend at which time we can test it again.

**Comment by sirbrillig** [comment]
Rebased and allowed refetching the backup toggle status on mount so that hopefully that will resolve the strange inaccurate data situations we've seen (which I believe are caching issues connected to redux).

**Review by pottedmeat — APPROVED** [review]
Tests went great and code looks good.

Are there any concerns with how many requests the backup checks make? Once for every stored card every time the window focuses.

**Comment by sirbrillig** [comment]
> Are there any concerns with how many requests the backup checks make? Once for every stored card every time the window focuses.

Yeah, that was why I had previously used the already-existing data from the Redux state instead, but in testing (see comments above) that was sometimes inaccurate. I imagine that most people don't have that many payment methods, so it should probably be ok?

**Comment by sirbrillig** [comment]
Note: expired cards can still be marked as a backup:
Screen Shot 2021-11-24 at 10 53 24 AM

File: client/me/purchases/payment-methods/payment-method-backup-toggle.tsx [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+	const isBackup = data?.is_backup ?? false;
+	return (
+		<div className="payment-method-backup-toggle">
+			<CheckboxControl
+				checked={ isBackup }
+				onChange={ toggleIsBackup }
+				label={ translate( 'Use as backup. {{link}}Learn more{{/link}}', {
+					components: {
+						link: (
+							<ExternalLink
```
Comment (michaeldcain) [inline client/me/purchases/payment-methods/payment-method-backup-toggle.tsx]: Is there a reason we opted for `External Link` vs `InlineSupport` here? I think the latter is preferred for support documentation and leaves the user in the context of the current screen.

Comment (michaeldcain) [reply client/me/purchases/payment-methods/payment-method-backup-toggle.tsx]: I made this change in 74223d4. Feel free to revert if there was a reason to use the other.

**Comment by michaeldcain** [comment]
I made some styling edits (largely to mobile), and replaced the Learn More link with an inline support icon.

Screen Shot 2021-12-06 at 2 58 31 PM
Screen Shot 2021-12-06 at 2 57 34 PM

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7025962

Thank you @sirbrillig for including a screenshot in the description! This is really helpful for our translators.

**Comment by sirbrillig** [comment]
Minor adjustment: https://github.com/Automattic/wp-calypso/pull/58913

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7025962).

**Pull request merged**